Docker with platform support failed to build and publish

Our release process include GitHub action to build and publish lakefs image to ECR and Docker Hub.
Last update to enable per platform build and speedup the local build per architecture in our Dockerfile, broke the release process:

```
Step 1/20 : FROM --platform=$BUILDPLATFORM golang:1.17.8-alpine AS build
failed to parse platform : "" is an invalid component of "": platform specifier component must match "^[A-Za-z0-9_-]+$": invalid argument
Error: Process completed with exit code 1.
```

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -63,8 +63,8 @@ jobs:
       - name: Login to DockerHub
         uses: docker/login-action@v2
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
 
       - name: Build and puth lakectl
         uses: docker/build-push-action@v3

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -46,7 +46,7 @@ jobs:
 
       - name: Extract version
         shell: bash
-        run: echo "::set-output name=tag::0.83.0b"
+        run: echo "::set-output name=tag::0.83.0c"
         #run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
         id: version
 

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -3,7 +3,7 @@ on:
   release:
     types: [published]
   pull_request:
-  pull:
+  push:
 
 # These permissions are needed to interact with GitHub's OIDC Token endpoint.
 permissions:

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -2,7 +2,8 @@ name: Docker
 on:
   release:
     types: [published]
-  commit:
+  pull_request:
+  pull:
 
 # These permissions are needed to interact with GitHub's OIDC Token endpoint.
 permissions:

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -2,6 +2,7 @@ name: Docker
 on:
   release:
     types: [published]
+  commit:
 
 # These permissions are needed to interact with GitHub's OIDC Token endpoint.
 permissions:
@@ -23,7 +24,7 @@ jobs:
           node-version: '16.15.0'
 
       - name: Set up Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v2
         with:
           go-version: 1.17.8
         id: go
@@ -44,46 +45,59 @@ jobs:
 
       - name: Extract version
         shell: bash
-        run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
+        run: echo "::set-output name=tag::0.83.0"
+        #run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
         id: version
-
-      - name: Extract db schema version from ddl migration files
-        run: echo "::set-output name=dbschema_version::$( (cd pkg/ddl/ && ls -d *.up.sql) | sed -nE '$s/^0*([0-9]+).*/\1/p' )"
-        id: schema
 
       - name: Login to Amazon ECR
         id: login-ecr
         uses: aws-actions/amazon-ecr-login@v1
 
-      - name: Build and push to Amazon ECR
-        env:
-          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
-          ECR_REPOSITORY_LAKEFS: lakefs
-          ECR_REPOSITORY_LAKECTL: lakectl
-          DOCKER_BUILDKIT: '1'
-        run: |
-          docker build --target lakectl -t $ECR_REGISTRY/$ECR_REPOSITORY_LAKECTL:${{ steps.version.outputs.tag }} --build-arg VERSION=${{ steps.version.outputs.tag }} --label db-schema-version=${{ steps.schema.outputs.dbschema_version}} .
-          docker push $ECR_REGISTRY/$ECR_REPOSITORY_LAKECTL:${{ steps.version.outputs.tag }}
-          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY_LAKEFS:${{ steps.version.outputs.tag }} --build-arg VERSION=${{ steps.version.outputs.tag }} --label db-schema-version=${{ steps.schema.outputs.dbschema_version}} .
-          docker push $ECR_REGISTRY/$ECR_REPOSITORY_LAKEFS:${{ steps.version.outputs.tag }}
-      - name: Build and push lakectl to Docker hub
-        uses: docker/build-push-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and puth lakefs to Amazon ECR
+        uses: docker/build-push-action@v3
         with:
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
-          build_args: VERSION=${{ steps.version.outputs.tag }}
-          labels: db-schema-version=${{ steps.schema.outputs.dbschema_version}}
+          target: lakefs
+          push: true
+          build-args:
+            - VERSION=${{ steps.version.outputs.tag }}
+          tags:
+            - ${{ steps.login-ecr.outputs.registry }}/lakefs:${{ steps.version.outputs.tag }}
+
+      - name: Build and puth lakectl to Amazon ECR
+        uses: docker/build-push-action@v3
+        with:
           target: lakectl
-          repository: treeverse/lakectl
-          tags: ${{ steps.version.outputs.tag }},latest
+          build-args:
+            - VERSION=${{ steps.version.outputs.tag }}
+          tags:
+            - ${{ steps.login-ecr.outputs.registry }}/lakectl:${{ steps.version.outputs.tag }}
 
       - name: Build and push lakefs to Docker hub
-        uses: docker/build-push-action@v1
+        uses: docker/build-push-action@v3
         with:
+          target: lakefs
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
-          build_args: VERSION=${{ steps.version.outputs.tag }}
-          labels: db-schema-version=${{ steps.schema.outputs.dbschema_version}}
-          target: lakefs
+          build_args:
+            - VERSION=${{ steps.version.outputs.tag }}
           repository: treeverse/lakefs
-          tags: ${{ steps.version.outputs.tag }},latest
+          tags:
+            - ${{ steps.version.outputs.tag }}
+            - latest
+
+      - name: Build and push lakectl to Docker hub
+        uses: docker/build-push-action@v3
+        with:
+          target: lakectl
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          build_args:
+            - VERSION=${{ steps.version.outputs.tag }}
+          repository: treeverse/lakectl
+          tags:
+            - ${{ steps.version.outputs.tag }}
+            - latest
+

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -17,9 +17,9 @@ jobs:
     steps:
 
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
-      - name: Setup NodeJS
+      - name: Set up NodeJS
         uses: actions/setup-node@v3
         with:
           node-version: '16.15.0'
@@ -50,50 +50,44 @@ jobs:
         #run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
         id: version
 
-      - name: Login to Amazon ECR
-        id: login-ecr
-        uses: aws-actions/amazon-ecr-login@v1
+      #- name: Set up QEMU
+        #uses: docker/setup-qemu-action@v2
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
 
-      - name: Build and puth lakefs to Amazon ECR
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and puth lakectl
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          target: lakectl
+          push: true
+          #platforms: linux/amd64,linux/arm64,darwin/amd64,darwin/arm64
+          build-args: VERSION=${{ steps.version.outputs.tag }}
+          tags: |
+            ${{ steps.login-ecr.outputs.registry }}/lakectl:${{ steps.version.outputs.tag }}
+            treeverse/lakectl:${{ steps.version.outputs.tag }}
+            treeverse/lakectl:latest
+
+      - name: Build and puth lakefs
         uses: docker/build-push-action@v3
         with:
           context: .
           target: lakefs
           push: true
+          #platforms: linux/amd64,linux/arm64,darwin/amd64,darwin/arm64
           build-args: VERSION=${{ steps.version.outputs.tag }}
-          tags: ${{ steps.login-ecr.outputs.registry }}/lakefs:${{ steps.version.outputs.tag }}
-
-      - name: Build and puth lakectl to Amazon ECR
-        uses: docker/build-push-action@v3
-        with:
-          context: .
-          push: true
-          target: lakectl
-          build-args: VERSION=${{ steps.version.outputs.tag }}
-          tags: ${{ steps.login-ecr.outputs.registry }}/lakectl:${{ steps.version.outputs.tag }}
-
-      - name: Build and push lakefs to Docker hub
-        uses: docker/build-push-action@v3
-        with:
-          context: .
-          push: true
-          target: lakefs
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
-          build_args: VERSION=${{ steps.version.outputs.tag }}
-          tags: treeverse/lakefs:${{ steps.version.outputs.tag }},treeverse/lakefs:latest
-
-      - name: Build and push lakectl to Docker hub
-        uses: docker/build-push-action@v3
-        with:
-          context: .
-          push: true
-          target: lakectl
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
-          build_args: VERSION=${{ steps.version.outputs.tag }}
-          tags: treeverse/lakectl:${{ steps.version.outputs.tag }},treeverse/lakectl:latest
-
+          tags: |
+            ${{ steps.login-ecr.outputs.registry }}/lakefs:${{ steps.version.outputs.tag }}
+            treeverse/lakefs:${{ steps.version.outputs.tag }}
+            treeverse/lakefs:latest

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -66,7 +66,7 @@ jobs:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
 
-      - name: Build and puth lakectl
+      - name: Build and push lakectl
         uses: docker/build-push-action@v3
         with:
           context: .
@@ -79,7 +79,7 @@ jobs:
             treeverse/lakectl:${{ steps.version.outputs.tag }}
             treeverse/lakectl:latest
 
-      - name: Build and puth lakefs
+      - name: Build and push lakefs
         uses: docker/build-push-action@v3
         with:
           context: .

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -46,12 +46,12 @@ jobs:
 
       - name: Extract version
         shell: bash
-        run: echo "::set-output name=tag::0.83.0a"
+        run: echo "::set-output name=tag::0.83.0b"
         #run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
         id: version
 
-      #- name: Set up QEMU
-        #uses: docker/setup-qemu-action@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
@@ -74,7 +74,7 @@ jobs:
           context: .
           target: lakectl
           push: true
-          #platforms: linux/amd64,linux/arm64,darwin/amd64,darwin/arm64
+          platforms: linux/amd64,linux/arm64,darwin/amd64,darwin/arm64
           build-args: VERSION=${{ steps.version.outputs.tag }}
           tags: |
             ${{ steps.login-ecr.outputs.registry }}/lakectl:${{ steps.version.outputs.tag }}
@@ -87,7 +87,7 @@ jobs:
           context: .
           target: lakefs
           push: true
-          #platforms: linux/amd64,linux/arm64,darwin/amd64,darwin/arm64
+          platforms: linux/amd64,linux/arm64,darwin/amd64,darwin/arm64
           build-args: VERSION=${{ steps.version.outputs.tag }}
           tags: |
             ${{ steps.login-ecr.outputs.registry }}/lakefs:${{ steps.version.outputs.tag }}

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -46,7 +46,7 @@ jobs:
 
       - name: Extract version
         shell: bash
-        run: echo "::set-output name=tag::0.83.0"
+        run: echo "::set-output name=tag::0.83.0a"
         #run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
         id: version
 
@@ -55,6 +55,8 @@ jobs:
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
+        with:
+          buildkitd-flags: --debug
 
       - name: Login to Amazon ECR
         id: login-ecr

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -60,6 +60,7 @@ jobs:
       - name: Build and puth lakefs to Amazon ECR
         uses: docker/build-push-action@v3
         with:
+          context: .
           target: lakefs
           push: true
           build-args: VERSION=${{ steps.version.outputs.tag }}
@@ -68,6 +69,8 @@ jobs:
       - name: Build and puth lakectl to Amazon ECR
         uses: docker/build-push-action@v3
         with:
+          context: .
+          push: true
           target: lakectl
           build-args: VERSION=${{ steps.version.outputs.tag }}
           tags: ${{ steps.login-ecr.outputs.registry }}/lakectl:${{ steps.version.outputs.tag }}
@@ -75,6 +78,8 @@ jobs:
       - name: Build and push lakefs to Docker hub
         uses: docker/build-push-action@v3
         with:
+          context: .
+          push: true
           target: lakefs
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
@@ -84,6 +89,8 @@ jobs:
       - name: Build and push lakectl to Docker hub
         uses: docker/build-push-action@v3
         with:
+          context: .
+          push: true
           target: lakectl
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -2,8 +2,6 @@ name: Docker
 on:
   release:
     types: [published]
-  pull_request:
-  push:
 
 # These permissions are needed to interact with GitHub's OIDC Token endpoint.
 permissions:
@@ -46,8 +44,7 @@ jobs:
 
       - name: Extract version
         shell: bash
-        run: echo "::set-output name=tag::0.83.0c"
-        #run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
+        run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
         id: version
 
       - name: Set up QEMU

--- a/.github/workflows/docker-publish.yaml
+++ b/.github/workflows/docker-publish.yaml
@@ -62,19 +62,15 @@ jobs:
         with:
           target: lakefs
           push: true
-          build-args:
-            - VERSION=${{ steps.version.outputs.tag }}
-          tags:
-            - ${{ steps.login-ecr.outputs.registry }}/lakefs:${{ steps.version.outputs.tag }}
+          build-args: VERSION=${{ steps.version.outputs.tag }}
+          tags: ${{ steps.login-ecr.outputs.registry }}/lakefs:${{ steps.version.outputs.tag }}
 
       - name: Build and puth lakectl to Amazon ECR
         uses: docker/build-push-action@v3
         with:
           target: lakectl
-          build-args:
-            - VERSION=${{ steps.version.outputs.tag }}
-          tags:
-            - ${{ steps.login-ecr.outputs.registry }}/lakectl:${{ steps.version.outputs.tag }}
+          build-args: VERSION=${{ steps.version.outputs.tag }}
+          tags: ${{ steps.login-ecr.outputs.registry }}/lakectl:${{ steps.version.outputs.tag }}
 
       - name: Build and push lakefs to Docker hub
         uses: docker/build-push-action@v3
@@ -82,12 +78,8 @@ jobs:
           target: lakefs
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
-          build_args:
-            - VERSION=${{ steps.version.outputs.tag }}
-          repository: treeverse/lakefs
-          tags:
-            - ${{ steps.version.outputs.tag }}
-            - latest
+          build_args: VERSION=${{ steps.version.outputs.tag }}
+          tags: treeverse/lakefs:${{ steps.version.outputs.tag }},treeverse/lakefs:latest
 
       - name: Build and push lakectl to Docker hub
         uses: docker/build-push-action@v3
@@ -95,10 +87,6 @@ jobs:
           target: lakectl
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
-          build_args:
-            - VERSION=${{ steps.version.outputs.tag }}
-          repository: treeverse/lakectl
-          tags:
-            - ${{ steps.version.outputs.tag }}
-            - latest
+          build_args: VERSION=${{ steps.version.outputs.tag }}
+          tags: treeverse/lakectl:${{ steps.version.outputs.tag }},treeverse/lakectl:latest
 

--- a/.github/workflows/esti.yaml
+++ b/.github/workflows/esti.yaml
@@ -90,7 +90,7 @@ jobs:
           node-version: '16.15.0'
 
       - name: Setup Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v2
         with:
           go-version: 1.17.8
         id: go
@@ -515,7 +515,7 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Setup Go
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v2
         with:
           go-version: 1.17.8
         id: go

